app/controllers/questions_controller.rb

```diff
@@ -1,8 +1,36 @@
 class QuestionsController < ApplicationController
-  access instructor: [:edit_questionnaire], instructor_pending: [:edit_questionnaire], admin: :all
+  access instructor: [:edit_questionnaire, :add_question, :sort], instructor_pending: [:edit_questionnaire, :add_question, :sort], admin: :all
 
   def edit_questionnaire
     @questions = Question.where(task_id: params[:review_task_id])
+    @questions = @questions.order(:position)
     @review_task = ReviewTask.find(params[:review_task_id])
   end
+
+  def sort
+    # respond_to do |format|
+      params[:question].each_with_index do |id, index|
+        Question.where(id: id).update_all(position: index + 1)
+    #  end
+    end
+
+    head :ok
+  end
+
+  def add_question
+    respond_to do |format|
+      if Question.create(question_params)
```

Comment:
If rails style polymorphism is used, here we can automatically create TextualFeedbackQuestion or NominationQuestion object, instead of Question object. (We can then use Question as an "abstract class" in Java)